pipeline/mutate/mutator_id_token.go

```diff
@@ -208,5 +208,28 @@ func (a *MutatorIDToken) Config(config json.RawMessage) (*CredentialsIDTokenConf
 		c.TTL = "15m"
 	}
 
+	cost := int64(c.Cache.MaxCost)
+	if cost == 0 {
+		cost = 1 << 25
+	}
+
+	if a.tokenCache == nil || a.tokenCache.MaxCost() != cost {
```

Comment:
The `a.tokenCache.MaxCost() != cost` condition is mainly for unit tests, in order to be able to test other `max_cost` values on the cache config.

On prod envs, this should not happen, and we'll keep skipping this branch.